src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -854,3 +855,210 @@ const Gaffer::GraphComponent *GafferScene::EditScopeAlgo::attributeEditReadOnlyR
 
 	return nullptr;
 }
+
+// Set Membership
+// ==============
+
+namespace
+{
+
+static int g_addSetColumnIndex = 0;
+static int g_removeSetColumnIndex = 1;
+
+SceneProcessorPtr setMembershipProcessor()
+{
+	SceneProcessorPtr result = new SceneProcessor( "SetMembershipEdits" );
+
+	SpreadsheetPtr spreadsheet = new Spreadsheet;
+	result->addChild( spreadsheet );
+	spreadsheet->selectorPlug()->setValue( "${setMembership:set}" );
+	spreadsheet->rowsPlug()->addColumn( new StringVectorDataPlug( "add", Plug::Direction::In, new StringVectorData() ) );
+	spreadsheet->rowsPlug()->addColumn( new StringVectorDataPlug( "remove", Plug::Direction::In, new StringVectorData() ) );
+
+	PathFilterPtr addPathFilter = new PathFilter;
+	result->addChild( addPathFilter );
+	addPathFilter->pathsPlug()->setInput( spreadsheet->outPlug()->getChild<StringVectorDataPlug>( g_addSetColumnIndex ) );
+
+	PathFilterPtr removePathFilter = new PathFilter;
+	result->addChild( removePathFilter );
+	removePathFilter->pathsPlug()->setInput( spreadsheet->outPlug()->getChild<StringVectorDataPlug>( g_removeSetColumnIndex ) );
+
+	GafferScene::SetPtr addSet = new GafferScene::Set();
+	result->addChild( addSet );
+	addSet->inPlug()->setInput( result->inPlug() );
+	addSet->filterPlug()->setInput( addPathFilter->outPlug() );
+	addSet->namePlug()->setInput( spreadsheet->enabledRowNamesPlug() );
+	addSet->modePlug()->setValue( GafferScene::Set::Mode::Add );
+	addSet->setVariablePlug()->setValue( "setMembership:set" );
```

Comment:
We should use a DeleteContextVariables to prevent this variable from becoming visible outside of the processor. Might also be good to use a ContextMonitor in the unit tests to assert that we don't leak anything.

Reply:
I think fixed in 4bbbee4beffcbf48d00de396642fb6aacb5eb1a6

Just to check that I understand, it shouldn't currently be leaking a context variable into the scene, so this is a precaution. Is that correct, or is my test not formulated correctly?

I couldn't get the test to fail so looked at `Set` node and saw that the context variable is only present for the filter connected to the Set, not the upstream scene evaluation.

Reply:
Sorry Eric, I sent you on a wild goose chase with this one - your analysis of what happens internally in the Set node is correct. We don't need to do any extra context handling of our own, so we can just drop https://github.com/GafferHQ/gaffer/commit/e19f2cbdb28a7ffc4a0341eaeeeaff5000502203.

Reply:
Cool, dropped before squash down into 83f348a5d5dbd06b4cf0cea7a664918e93ffdad0